keyless/pepper/common/src/vuf/bls12381_g1_bls.rs

```diff
@@ -14,19 +18,131 @@ use ark_std::{
     rand::{CryptoRng, RngCore},
     UniformRand,
 };
+use blstrs::{self, Compress};
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+use serde_big_array::BigArray;
 use std::ops::Mul;
 
 pub struct Bls12381G1Bls {}
 
-pub static DST: &[u8] = b"APTOS_PEPPER_SERVICE_BLS12381_VUF_DST";
+pub static DST: &[u8] = b"APTOS_PEPPER_BLS12381_VUF_DST";
+
+pub static PINKAS_DST: &[u8] = b"APTOS_PEPPER_PINKAS_VUF_DST";
+
+pub static PINKAS_SECRET_KEY_BASE: &[u8] = b"APTOS_PEPPER_PINKAS_VUF_SECRET_KEY_BASE";
+
+pub static PINKAS_SECRET_KEY_BASE_AFFINE: Lazy<G2Affine> =
+    Lazy::new(|| Bls12381G1Bls::hash_to_g2(PINKAS_SECRET_KEY_BASE, PINKAS_DST));
+
+pub static PINKAS_SECRET_KEY_BASE_G2: Lazy<blstrs::G2Projective> =
+    Lazy::new(|| {
+        let mut g2_bytes = vec![];
+        PINKAS_SECRET_KEY_BASE_AFFINE
+            .serialize_compressed(&mut g2_bytes)
+            .map_err(|e| {
+                anyhow!("Bls12381G1Bls::eval failed with output serialization error: {e}")
+            }).unwrap();
+        let b2: [u8; 96] = g2_bytes.clone()[0..96]
+            .try_into().unwrap();
+        blstrs::G2Projective::from_compressed(&b2).unwrap()
+    });
+
+#[derive(Serialize, Deserialize, CryptoHasher, BCSCryptoHash)]
+pub struct PinkasPepper {
+    #[serde(with = "BigArray")]
+    pub bytes: [u8; 288],
+}
+
+impl PinkasPepper {
+
+    pub fn from_affine_bytes(input: &[u8]) -> anyhow::Result<PinkasPepper> {
+        let g1 = G1Affine::deserialize_compressed(input)?;
+        let mut g1_bytes = vec![];
+        g1.serialize_compressed(&mut g1_bytes).map_err(|e| {
+            anyhow!("Bls12381G1Bls::eval failed with output serialization error: {e}")
+        })?;
+        let b1: [u8; 48] = g1_bytes.clone()[0..48]
+            .try_into()
+            .expect("Expected 48 bytes");
```

Comment:
Are lines 60 through 67 simply converting a `[u8]` into a `[u8; 48]`? I'm confused by the reserialization...

Reply:
i printed it out at one point